spec/tbd_tests_spec.rb

```diff
@@ -2425,4 +2425,1277 @@
+    expect(surface[:windows]["w1"][:points].size).to eq(3)
+    expect(surface[:windows]["w2"][:points].size).to eq(4)
+    expect(surface[:windows]["w3"][:points].size).to eq(3)
+
+    ptz = surface[:windows]["w3"][:points]
+    expect(ptz.size).to eq(3)
+    vec = OpenStudio::Point3dVector.new
+
+    ptz.each { |o| vec << OpenStudio::Point3d.new(o.x, o.y, o.z) }
+
+    # This window would have 2 shared edges (@right angle) with the parent.
+    expect(vec[0].x).to be_within(TOL).of( 8.52)
+    expect(vec[0].y).to be_within(TOL).of( 0.00)
+    expect(vec[0].z).to be_within(TOL).of(10.00)
+    expect(vec[1].x).to be_within(TOL).of(10.00)
+    expect(vec[1].y).to be_within(TOL).of( 0.00)
+    expect(vec[1].z).to be_within(TOL).of( 8.52)
+    expect(vec[2].x).to be_within(TOL).of(10.00)
+    expect(vec[2].y).to be_within(TOL).of( 0.00)
+    expect(vec[2].z).to be_within(TOL).of(10.00)
+
+
+    # --- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- --- #
+    # Repeat exercise, with parent surface & subsurfaces rotated 120 (CW).
+    # (i.e., negative coordinates, Y-axis coordinates, etc.)
+    model = OpenStudio::Model::Model.new
+    space = OpenStudio::Model::Space.new(model)
+    space.setName("Space")
+
+    # All subsurfaces are Simple Glazing constructions.
+    fen = OpenStudio::Model::Construction.new(model)
+    fen.setName("FD fen")
+
+    glazing = OpenStudio::Model::SimpleGlazing.new(model)
+    glazing.setName("FD glazing")
+    expect(glazing.setUFactor(2.0)).to be true
+
+    layers = OpenStudio::Model::MaterialVector.new
+    layers << glazing
+    expect(fen.setLayers(layers)).to be true
+
+    vec  = OpenStudio::Point3dVector.new
+    vec << OpenStudio::Point3d.new(  0.00,  0.00, 10.00)
+    vec << OpenStudio::Point3d.new(  0.00,  0.00,  0.00)
+    vec << OpenStudio::Point3d.new( -5.00, -8.66,  0.00)
+    vec << OpenStudio::Point3d.new( -5.00, -8.66, 10.00)
+    dad  = OpenStudio::Model::Surface.new(vec, model)
+    dad.setName("dad")
+    expect(dad.setSpace(space)).to be true
+
+    vec  = OpenStudio::Point3dVector.new
+    vec << OpenStudio::Point3d.new( -1.00, -1.73,  8.00)
+    vec << OpenStudio::Point3d.new( -0.50, -0.87,  6.00)
+    vec << OpenStudio::Point3d.new( -2.00, -3.46,  9.00)
+    w1   = OpenStudio::Model::SubSurface.new(vec, model)
+    w1.setName("w1")
+    expect(w1.setSubSurfaceType("FixedWindow")).to be true
+    expect(w1.setSurface(dad)).to be true
+    expect(w1.setConstruction(fen)).to be true
+    expect(w1.uFactor).to_not be_empty
+    expect(w1.uFactor.get).to be_within(0.1).of(2.0)
+    expect(w1.netArea  ).to be_within(TOL).of(1.50)
+    expect(w1.grossArea).to be_within(TOL).of(1.50)
+
+    vec  = OpenStudio::Point3dVector.new
+    vec << OpenStudio::Point3d.new( -3.50, -6.060,  4.00)
+    vec << OpenStudio::Point3d.new( -2.00, -3.460,  1.00)
+    vec << OpenStudio::Point3d.new( -4.00, -6.930,  2.00)
+    vec << OpenStudio::Point3d.new( -4.50, -7.795,  3.00)
```

Comment:
A recent change in OSut's basic polygon tests triggered failures in this oldish test (TBD Test repo). These XYZ coordinates are SU-reported, with a precision of ~1cm. Fixes are limited to small adjustments (a few millimetres here and there). May drop these test variations altogether in the future :/